test/parachain/tests/integration_test.rs

```diff
@@ -0,0 +1,146 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+// This test needs --release to work
+#![cfg(not(debug_assertions))]
+
+use assert_cmd::cargo::cargo_bin;
+use std::{
+	convert::TryInto, process::{Child, Command, Stdio}, thread, time::Duration, fs, path::PathBuf,
+	env, net, io,
+};
+use tempfile::tempdir;
+use jsonrpc_client_transports::transports::ws;
+use jsonrpc_client_transports::RpcChannel;
+use url::Url;
+
+static POLKADOT_ARGS: &[&str] = &["polkadot", "--chain=res/polkadot_chainspec.json"];
+
+jsonrpsee::rpc_api! {
+	Health {
+		/// Test
+		fn system_name(foo: String, bar: i32) -> String;
+
+		fn test_notif(foo: String, bar: i32);
+
+		/// Test2
+		#[rpc(method = "foo")]
+		fn system_name2() -> String;
+	}
+
+	System {
+		fn test_foo() -> String;
+	}
+}
+
+// Adapted from
+// https://github.com/rust-lang/cargo/blob/485670b3983b52289a2f353d589c57fae2f60f82/tests/testsuite/support/mod.rs#L507
+fn target_dir() -> PathBuf {
+	env::current_exe()
+		.ok()
+		.map(|mut path| {
+			path.pop();
+			if path.ends_with("deps") {
+				path.pop();
+			}
+			path
+		})
+		.unwrap()
+}
+
+struct ProcessCleanUp<'a>(&'a mut Child);
+
+impl<'a> Drop for ProcessCleanUp<'a> {
+	fn drop(&mut self) {
+		let _ = self.0.kill();
+		let _ = self.0.wait();
+	}
+}
+
+fn wait_for_tcp<A: net::ToSocketAddrs>(address: A) -> io::Result<()> {
+	let mut tries = 10;
+
+	loop {
+		tries -= 1;
+
+		match net::TcpStream::connect(&address) {
+			Ok(_) => break Ok(()),
+			Err(err) if tries == 0 => break Err(err),
+			_ => thread::sleep_ms(1000),
+		}
+	}
+}
+
+#[test]
+//#[tokio::test]
+fn integration_test() {
+	// starts Alice
+	let polkadot_alice_dir = tempdir().unwrap();
+	let mut polkadot_alice = Command::new(cargo_bin("cumulus-test-parachain-collator"))
+		.stdout(Stdio::piped())
+		.stderr(Stdio::piped())
+		.args(POLKADOT_ARGS)
+		.arg("--base-path")
+		.arg(polkadot_alice_dir.path())
+		.arg("--alice")
+		.spawn()
+		.unwrap();
+	let polkadot_alice_clean_up = ProcessCleanUp(&mut polkadot_alice);
+	wait_for_tcp("127.0.0.1:9933").unwrap();
+
+	// starts Bob
+	let polkadot_bob_dir = tempdir().unwrap();
+	let mut polkadot_bob = Command::new(cargo_bin("cumulus-test-parachain-collator"))
+		.stdout(Stdio::piped())
+		.stderr(Stdio::piped())
+		.args(POLKADOT_ARGS)
+		.arg("--base-path")
+		.arg(polkadot_bob_dir.path())
+		.arg("--bob")
+		.spawn()
+		.unwrap();
+	let polkadot_bob_clean_up = ProcessCleanUp(&mut polkadot_bob);
```

Comment:
Should Bob's node have some arguments to set a different port? Should we `wait_for_tcp` for Bob's port?

Reply:
Not for my test based on the manual example but sure you can do whatever you want for your use case.